Documentation/building/versioning.md

```diff
@@ -0,0 +1,41 @@
+Package and File Versioning
+===========================
+
+We have now added package versioning and assembly file versioning when building the repo. This versioning works in both Windows and non-Windows platforms. Also, this versioning applies for both the managed and native libraries we produce. We had a few constrains to take into account, for example:
+- Version needed to be higher than the versions that we have already shipped.
+- There needs to be an ability to have multiple versions per day.
+- Versions need to be always increasing.
+- Version needs to be lower than [SHRT_MAX](https://www.google.com/?ion=1&espv=2#q=max%20short%20int) since the version is used as assembly file version which has that constrain.
+- (Nice to have) Version needs to be reproducible.
```

Comment:
The problem is that today it is not reproducible entirely. This is because only the BuildNumberMajor is reproducible, but the minor is still calculated by the build system, so I can mark that as a requirement but the problem is that we do not fulfill this completely yet.